tests/bluetooth/audio/ascs/src/test_ase_state_transition.c

```diff
@@ -0,0 +1,385 @@
+	    test_ase_state_transition_before, test_ase_state_transition_after,
+	    test_ase_state_transition_teardown);
+
+ZTEST_F(test_sink_ase_state_transition, test_client_idle_to_codec_configured)
+{
+	struct bt_bap_stream *stream = &fixture->stream;
+	struct bt_conn *conn = &fixture->conn;
+	uint8_t ase_id = fixture->ase.id;
+
+	Z_TEST_SKIP_IFNDEF(CONFIG_BT_ASCS_ASE_SNK);
+
+	test_ase_control_client_config_codec(conn, ase_id, stream);
+	expect_bt_bap_unicast_server_cb_config_called_once(conn, EMPTY, BT_AUDIO_DIR_SINK, EMPTY);
+	expect_bt_bap_stream_ops_configured_called_once(stream, EMPTY);
+}
+
+ZTEST_F(test_sink_ase_state_transition, test_client_codec_configured_to_qos_configured)
+{
+	struct bt_bap_stream *stream = &fixture->stream;
+	struct bt_conn *conn = &fixture->conn;
+	uint8_t ase_id = fixture->ase.id;
+
+	Z_TEST_SKIP_IFNDEF(CONFIG_BT_ASCS_ASE_SNK);
+
+	/* Preamble */
+	test_ase_control_client_config_codec(conn, ase_id, stream);
+
+	test_ase_control_client_config_qos(conn, ase_id);
+	expect_bt_bap_unicast_server_cb_qos_called_once(stream, EMPTY);
+	expect_bt_bap_stream_ops_qos_set_called_once(stream);
+}
+
+ZTEST_F(test_sink_ase_state_transition, test_client_qos_configured_to_enabling)
+{
+	struct bt_bap_stream *stream = &fixture->stream;
+	struct bt_conn *conn = &fixture->conn;
+	uint8_t ase_id = fixture->ase.id;
+
+	Z_TEST_SKIP_IFNDEF(CONFIG_BT_ASCS_ASE_SNK);
+
+	/* Preamble */
+	test_ase_control_client_config_codec(conn, ase_id, stream);
+	test_ase_control_client_config_qos(conn, ase_id);
+
+	test_ase_control_client_enable(conn, ase_id);
+	expect_bt_bap_unicast_server_cb_enable_called_once(stream, EMPTY, EMPTY);
+	expect_bt_bap_stream_ops_enabled_called_once(stream);
+}
+
+ZTEST_F(test_sink_ase_state_transition, test_client_enabling_to_qos_configured)
+{
+	struct bt_bap_stream *stream = &fixture->stream;
+	struct bt_conn *conn = &fixture->conn;
+	uint8_t ase_id = fixture->ase.id;
+
+	Z_TEST_SKIP_IFNDEF(CONFIG_BT_ASCS_ASE_SNK);
+
+	/* Preamble */
+	test_ase_control_client_config_codec(conn, ase_id, stream);
+	test_ase_control_client_config_qos(conn, ase_id);
+	test_ase_control_client_enable(conn, ase_id);
+
+	/* Reset the bap_stream_ops.qos_set callback that is expected to be called again */
+	mock_bap_stream_qos_set_cb_reset();
+
+	test_ase_control_client_disable(conn, ase_id);
+	expect_bt_bap_unicast_server_cb_disable_called_once(stream);
+	expect_bt_bap_stream_ops_qos_set_called_once(stream);
+}
+
+ZTEST_F(test_sink_ase_state_transition, test_client_qos_configured_to_releasing)
+{
+	struct bt_bap_stream *stream = &fixture->stream;
+	struct bt_conn *conn = &fixture->conn;
+	uint8_t ase_id = fixture->ase.id;
+
+	Z_TEST_SKIP_IFNDEF(CONFIG_BT_ASCS_ASE_SNK);
+
+	/* Preamble */
+	test_ase_control_client_config_codec(conn, ase_id, stream);
+	test_ase_control_client_config_qos(conn, ase_id);
+	test_ase_control_client_enable(conn, ase_id);
+	test_ase_control_client_disable(conn, ase_id);
```

Comment:
Hmm, `to_releasing` can basically be done in any state, right? What this really does is `enabling`  (or rather `disabling`) to release, right?

Reply:
Actually disable for ASE in `enabling` state moves it to `qos_configured` state.
I think this test case shall look rather like:
```
	/* Preamble */
	test_ase_control_client_config_codec(conn, ase_id, stream);
	test_ase_control_client_config_qos(conn, ase_id);

	test_ase_control_client_release(conn, ase_id);
```
I'll change it